mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -1242,7 +1243,7 @@ def EmitC_SubscriptOp : EmitC_Op<"subscript", []> {
       EmitC_OpaqueType,
       EmitC_PointerType]>,
     "the value to subscript">:$value,
-    Variadic<EmitCType>:$indices);
+    Variadic<AnyTypeOf<[EmitCType, EmitC_SignedSizeT, EmitC_SizeT]>>:$indices);
```

Comment:
Can we extend `EmitCType` to include EmitC_SignedSizeT, EmitC_SizeT instead of changing it here?

Reply:
Woah, that was quick! I actually want to do a couple things before submitting this: 
* introduce a index type function check (like we do for EmitC integers), (so this will answer this comment),
* make use of the `EmitC_SizeType` abstract type (which `size_t` and `ssize_t` derive from),
* factor the signed/unsigned conversions in case of type mismatch into a function (this code is repeated many times across the ArithToEmitC pass).